ext/opentelemetry-ext-jinja2/src/opentelemetry/ext/jinja2/__init__.py

```diff
@@ -0,0 +1,134 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+
+Usage
+-----
+
+The OpenTelemetry ``jinja2`` integration traces templates loading, compilation and rendering.
+
+.. code-block:: python
+
+Instrumentation example::
+
+    from opentelemetry.ext.jinja2 import Jinja2Instrumentor
+    Jinja2Instrumentor().instrument()  # This needs to be executed before importing jinja2
+    from jinja2 import Environment, FileSystemLoader
+
+    env = Environment(
+        loader=FileSystemLoader("templates")
+    )
+    template = env.get_template('mytemplate.html')
+
+API
+---
+"""
+
+import logging
+
+import jinja2
+from wrapt import ObjectProxy
+from wrapt import wrap_function_wrapper as _wrap
+
+from opentelemetry import trace
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.jinja2.version import __version__
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TEMPLATE_NAME = "<memory>"
+
+
+def _wrap_render(wrapped, instance, args, kwargs):
+    """Wrap `Template.render()` or `Template.generate()`
+    """
+    template_name = instance.name or DEFAULT_TEMPLATE_NAME
+    tracer = trace.get_tracer(__name__, __version__)
+    with tracer.start_as_current_span(
+        "jinja2.render", kind=trace.SpanKind.INTERNAL
+    ) as span:
+        try:
+            return wrapped(*args, **kwargs)
+        finally:
+            span.set_attribute("component", "template")
+            span.set_attribute("jinja2.template_name", template_name)
+
+
+def _wrap_compile(wrapped, _, args, kwargs):
+    if len(args) > 1:
+        template_name = args[1]
+    else:
+        template_name = kwargs.get("name", DEFAULT_TEMPLATE_NAME)
+
+    tracer = trace.get_tracer(__name__, __version__)
+    with tracer.start_as_current_span(
+        "jinja2.compile", kind=trace.SpanKind.INTERNAL
+    ) as span:
+        try:
+            return wrapped(*args, **kwargs)
+        finally:
+            span.set_attribute("component", "template")
+            span.set_attribute("jinja2.template_name", template_name)
```

Comment:
Template engines are widely used across different languages. Have you considered bringing up a spec PR for semantic conventions for templating engines? I think we should aim for cross-language cross-framework consistency here. E.g. maybe `templating.engine=jinja2`, `templating.template=FILENAME` would be good candidates?

As long as we don't have such conventions, using `jinja2` as prefix is however probably the most name-clash-safe way.

Also, please set these attributes already when creating the span, so that they are available to samplers.

Reply:
Certainly agree about a spec PR and can start a conversation about that as a follow-up from this PR.